### PR TITLE
Feature/Remote Control Radio Parameter Update 

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
### This implementation is based on [RC new modules](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0099-new-remote-control-modules-and-parameters.md), so this PR should be merged after https://github.com/smartdevicelink/sdl_core/pull/2207

**PR for review https://github.com/smartdevicelink/sdl_core/pull/2261**

Technical task : #2162 
This PR is **ready** for review.

This pull request contains the implementation of Remote Control Radio Parameter Update 
(proposal: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0160-rc-radio-parameter-update.md)

Test scripts: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1927
For testing use the following test set:
https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/feature/rc_radio_parameter_update/test_sets/rc_radio_parameter_update.txt

[HMI integration guidelines](https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/pull/106)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by AStasiuk** [comment]
@yang1070 could you please approve the original PR #2264 as well?

**Comment by LitvinenkoIra** [comment]
@JackLivio feature/rc_radio_parameter_update rebased on latest develop

**Comment by yang1070** [comment]
@LitvinenkoIra @LuxoftAKutsan This PR also need a rebase after merge of 2207.

File: src/components/interfaces/HMI_API.xml [hunk 1]
```diff
@@ -1853,10 +1853,13 @@
    </param>
    <param name="rdsData" type="Common.RdsData" mandatory="false">
    </param>
-   <param name="availableHDs" type="Integer" minvalue="1" maxvalue="3" mandatory="false">
+   <param name="hdRadioEnable" type="Boolean" mandatory="false">
```
Comment (yang1070) [inline src/components/interfaces/HMI_API.xml]: since 5.0.0

File: src/components/interfaces/HMI_API.xml [hunk 2]
```diff
@@ -1853,10 +1853,13 @@
...
+     <description> True if the hd radio is on, false is the radio is off</description>
+   </param>
+   <param name="availableHDs" type="Integer" minvalue="1" maxvalue="7" mandatory="false">
```
Comment (yang1070) [inline src/components/interfaces/HMI_API.xml]: history?

File: src/components/interfaces/HMI_API.xml [hunk 3]
```diff
@@ -1853,10 +1853,13 @@
...
      <description>number of HD sub-channels if available</description>
    </param>
-   <param name="hdChannel" type="Integer" minvalue="1" maxvalue="3" mandatory="false">
+   <param name="hdChannel" type="Integer" minvalue="1" maxvalue="7" mandatory="false">
```
Comment (yang1070) [inline src/components/interfaces/HMI_API.xml]: history

File: src/components/interfaces/HMI_API.xml [hunk 4]
```diff
@@ -1939,6 +1942,18 @@
          True: Available, False: Not Available, Not present: Not Available.
        </description>
    </param>
+   <param name="hdRadioEnableAvailable" type="Boolean" mandatory="false">
```
Comment (yang1070) [inline src/components/interfaces/HMI_API.xml]: since 5.0.0

File: src/components/interfaces/HMI_API.xml [hunk 5]
```diff
@@ -1939,6 +1942,18 @@
...
+     <description>
+       Availability of the control of enable/disable HD radio.
+       True: Available, False: Not Available, Not present: Not Available.
+     </description>
+   </param>
+   <param name="siriusxmRadioAvailable" type="Boolean" mandatory="false">
```
Comment (yang1070) [inline src/components/interfaces/HMI_API.xml]: since 5.0.0

File: src/components/interfaces/MOBILE_API.xml [hunk 6]
```diff
@@ -3034,10 +3034,13 @@
         </param>
         <param name="rdsData" type="RdsData" mandatory="false">
         </param>
-        <param name="availableHDs" type="Integer" minvalue="1" maxvalue="3" mandatory="false">
+        <param name="hdRadioEnable" type="Boolean" mandatory="false">
```
Comment (yang1070) [inline src/components/interfaces/MOBILE_API.xml]: since 5.0

File: src/components/interfaces/MOBILE_API.xml [hunk 7]
```diff
@@ -3034,10 +3034,13 @@
...
+            <description> True if the hd radio is on, false is the radio is off</description>
+        </param>
+        <param name="availableHDs" type="Integer" minvalue="1" maxvalue="7" mandatory="false">
```
Comment (yang1070) [inline src/components/interfaces/MOBILE_API.xml]: since 5.0 and history

File: src/components/interfaces/MOBILE_API.xml [hunk 8]
```diff
@@ -3034,10 +3034,13 @@
...
             <description>number of HD sub-channels if available</description>
         </param>
-        <param name="hdChannel" type="Integer" minvalue="1" maxvalue="3" mandatory="false">
+        <param name="hdChannel" type="Integer" minvalue="1" maxvalue="7" mandatory="false">
```
Comment (yang1070) [inline src/components/interfaces/MOBILE_API.xml]: since 5.0 and history

**Comment by yang1070** [comment]
@theresalech @jacobkeeler Ford approves this change. It need to be merged after #2207

**Pull request merged**